### DISCUSSION TITLE
面板修改端口后进不去

### DISCUSSION THREAD

**Discussion by YiHooong** [1Panel-dev/1Panel#7131] [Q&A]
### 联系方式

2066203837@qq.com

### 1Panel 版本

1.10.20-lts

### 问题描述

修改了一次面板端口后就进不去面板了 面板安全入口是关着的 能确认修改的端口没有被占用 
之前也碰到过几次 有的时候是升级导致莫名其妙进不去 有的时候换端口导致进不去 才想起来提issue 本人比较小白抱歉 
下面是1pctl status

lines 1-20/20 (END)...skipping...
● 1panel.service - 1Panel, a modern open source linux panel
     Loaded: loaded (/etc/systemd/system/1panel.service; enabled; preset: enabled)
     Active: active (running) since Sun 2024-11-17 15:24:52 UTC; 1min 8s ago
   Main PID: 3582 (1panel)
      Tasks: 7 (limit: 18849)
     Memory: 46.3M
        CPU: 478ms
     CGroup: /system.slice/1panel.service
             └─3582 /usr/bin/1panel

11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] POST   /api/v1/alerts/logs/sync  --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).SyncSmsInfo-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] POST   /api/v1/alerts/logs/clean --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).CleanAlertLogs-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] POST   /api/v1/alerts/logs/sync/all --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).SyncSmsAll-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] GET    /api/v1/alerts/clams/list --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).GetClams-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] POST   /api/v1/alerts/cronjob/list --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).GetCronJobs-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [GIN-debug] POST   /api/v1/alerts/sms        --> github.com/1Panel-dev/1Panel/backend/xpack/app/api/v1.(*XpackApi).SyncSmsCount-fm (12 handlers)
11月 17 15:24:55 Debian 1panel[3582]: [2024-11-17 15:24:55] [INFO] listen at http://0.0.0.0:6666 [tcp4]
11月 17 15:24:56 Debian 1panel[3582]: [2024-11-17 15:24:56] [INFO] [AppStore] download file from https://apps-assets.fit2cloud.com/stable/1panel.json.zip
11月 17 15:24:56 Debian 1panel[3582]: [2024-11-17 15:24:56] [INFO] The App Store is at the latest version
11月 17 15:25:07 Debian 1panel[3582]: [2024-11-17 15:25:07] [ERROR] load remote time with [pool.ntp.org] failed, err: failed to read server response: read udp 192.168.1.20:37139->198.18.0.137:123: i/o timeout

### 重现步骤

关闭面板安全入口 修改端口

**Comment by ssongliu** [comment]
检查一下防火墙以及端口占用情况
netstat -tunlp | grep 6666
ufw status verbose

**Comment by YiHooong** [comment]
> 检查一下防火墙以及端口占用情况 netstat -tunlp | grep 6666 ufw status verbose
root@Debian:~# netstat -tunlp | grep 6666
tcp6       0      0 :::6666                 :::*                    LISTEN      779323/1panel       
root@Debian:~# ufw status verbose
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), deny (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
22/tcp                     ALLOW IN    192.168.1.0/24            
80/tcp                     ALLOW IN    Anywhere                  
3000                       ALLOW IN    Anywhere                  
5678/tcp                   ALLOW IN    Anywhere                  
4567/tcp                   ALLOW IN    Anywhere                  
5000                       ALLOW IN    Anywhere                  
40000/tcp                  ALLOW IN    Anywhere                  
6666/tcp                   ALLOW IN    Anywhere                  
80/tcp (v6)                ALLOW IN    Anywhere (v6)             
3000 (v6)                  ALLOW IN    Anywhere (v6)             
5678/tcp (v6)              ALLOW IN    Anywhere (v6)             
4567/tcp (v6)              ALLOW IN    Anywhere (v6)             
5000 (v6)                  ALLOW IN    Anywhere (v6)             
40000/tcp (v6)             ALLOW IN    Anywhere (v6)             
6666/tcp (v6)              ALLOW IN    Anywhere (v6)             

root@Debian:~# 
看着都没问题 但是就是进不去

**Comment by wanghe-fit2cloud** [comment]
1、`netstat -ntlp` 看一下端口绑定情况并截图
2、`curl 127.0.0.1:端口` 看一下结果并提供截图

**Comment by YiHooong** [comment]
> 1、`netstat -ntlp` 看一下端口绑定情况并截图 2、`curl 127.0.0.1:端口` 看一下结果并提供截图
![{1FB281DD-60A2-4193-81C2-531B694E1516}](https://github.com/user-attachments/assets/25bb252f-6726-4549-b074-f96eed5ce2ce)
curl太长了就直接复制过来吧

root@Debian:~# curl 127.0.0.1:6666
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <meta name="robots" content="noindex,nofollow" />
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <title>loading...</title>
      <script type="module" crossorigin src="/assets/js/index-BYC_zW2c.js"></script>
      <link rel="stylesheet" crossorigin href="/assets/css/index--enzMEcn.css">
    </head>
    <body>
        <div id="app">
            <style>
                html,
                body,
                #app {
                    width: 100%;
                    height: 100%;
                    padding: 0;
                    margin: 0;
                }
                .first-loading-wrap {
                    display: flex;
                    flex-direction: column;
                    align-items: center;
                    justify-content: center;
                    width: 100%;
                    height: 100%;
                }
                .first-loading-wrap > h1 {
                    font-size: 128px;
                }
                .first-loading-wrap .loading-wrap {
                    display: flex;
                    align-items: center;
                    justify-content: center;
                    padding: 98px;
                }
                .dot {
                    position: relative;
                    box-sizing: border-box;
                    display: inline-block;
                    width: 32px;
                    height: 32px;
                    font-size: 32px;
                    transform: rotate(45deg);
                    animation: ant-rotate 1.2s infinite linear;
                }
                .dot i {
                    position: absolute;
                    display: block;
                    width: 14px;
                    height: 14px;
                    background-color: #1890ff;
                    border-radius: 100%;
                    opacity: 0.3;
                    transform: scale(0.75);
                    transform-origin: 50% 50%;
                    animation: ant-spin-move 1s infinite linear alternate;
                }
                .dot i:nth-child(1) {
                    top: 0;
                    left: 0;
                }
                .dot i:nth-child(2) {
                    top: 0;
                    right: 0;
                    animation-delay: 0.4s;
                }
                .dot i:nth-child(3) {
                    right: 0;
                    bottom: 0;
                    animation-delay: 0.8s;
                }
                .dot i:nth-child(4) {
                    bottom: 0;
                    left: 0;
                    animation-delay: 1.2s;
                }
                @keyframes ant-rotate {
                    to {
                        transform: rotate(405deg);
                    }
                }
                @keyframes ant-spin-move {
                    to {
                        opacity: 1;
                    }
                }
            </style>
            <div class="first-loading-wrap">
                <div class="loading-wrap">
                    <span class="dot dot-spin">
                        <i></i>
                        <i></i>
                        <i></i>
                        <i></i>
                    </span>
                </div>
            </div>
        </div>
    </body>
</html>
root@Debian:~# 

这个问题好像在我这可以完美复现 我卸载后删除关于所有1panel的内容 重装之后改端口就进不去

**Comment by wanghe-fit2cloud** [comment]
使用 `1pctl listen-ip ipv4` 命令，将 `6666` 端口绑定到 IPv4 网络，然后在浏览器尝试一下通过 `http://IPv4地址:6666` 访问。

**Comment by YiHooong** [comment]
> 使用 `1pctl listen-ip ipv4` 命令，将 `6666` 端口绑定到 IPv4 网络，然后在浏览器尝试一下通过 `http://IPv4地址:6666` 访问。
root@Debian:~# 1pctl listen-ip ipv4
● 1panel.service - 1Panel, a modern open source linux panel
     Loaded: loaded (/etc/systemd/system/1panel.service; enabled; preset: enabled)
     Active: active (running) since Tue 2024-11-19 14:14:35 UTC; 8ms ago
   Main PID: 989093 (1panel)
      Tasks: 5 (limit: 18849)
     Memory: 1.6M
        CPU: 7ms
     CGroup: /system.slice/1panel.service
             └─989093 /usr/bin/1panel

11月 19 14:14:35 Debian systemd[1]: Started 1panel.service - 1Panel, a modern open >
lines 1-11/11 (END)
修改后还是不是不行 重启面板也是进不去

**Comment by wanghe-fit2cloud** [comment]
执行命令 `netstat -ntlp`，截图看一下端口是不是绑定到 IPv4

**Comment by YiHooong** [comment]
> 执行命令 `netstat -ntlp`，截图看一下端口是不是绑定到 IPv4
![Screenshot 2024-11-19 22 20 57](https://github.com/user-attachments/assets/b38f32b6-d806-4221-8142-65766abafed5)
看起来是在ipv4上的

**Comment by wanghe-fit2cloud** [comment]
服务运行和端口绑定均正常，且通过 `curl 127.0.0.1:端口` 测试也正常，说明问题可能出在 1Panel 所在服务器的网络配置。你可以重点检查一下主机防火墙和安全组设置，确保没有阻塞该端口。
